cmd/chartsvc/handler.go

```diff
@@ -200,6 +204,90 @@ func getChartVersionValues(w http.ResponseWriter, req *http.Request, params Para
 	w.Write([]byte(files.Values))
 }
 
+func isChartContained(chartData *models.Chart, chartMetadata *chart.Metadata) bool {
+	eq := true
+	// Compare simple fields
+	if chartData.Description != chartMetadata.Description ||
+		chartData.Home != chartMetadata.Home ||
+		chartData.Icon != chartMetadata.Icon {
+		eq = false
+	}
+	// Compare array of key words (but it may be empty)
+	if len(chartData.Keywords) > 0 && len(chartMetadata.Keywords) > 0 {
+		if !reflect.DeepEqual(chartData.Keywords, chartMetadata.Keywords) {
+			eq = false
+		}
+	}
+	// Compare array of maintainers (but it may be empty)
+	if len(chartData.Maintainers) > 0 && len(chartMetadata.Maintainers) > 0 {
+		// Maintainers format is different (pointers versus structs)
+		c2MaintainerArray := []chart.Maintainer{}
+		for _, m := range chartMetadata.Maintainers {
+			c2MaintainerArray = append(c2MaintainerArray, *m)
+		}
+		if !reflect.DeepEqual(chartData.Maintainers, c2MaintainerArray) {
+			eq = false
+		}
+	}
+	if eq {
+		// Charts matches, look if the version exists
+		for _, v := range chartData.ChartVersions {
+			if v.Version == chartMetadata.Version && v.AppVersion == chartMetadata.AppVersion {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// resolveRepo returns the repo that contains the given chart and the latest version found
+func resolveRepo(w http.ResponseWriter, req *http.Request) {
+	db, closer := dbSession.DB()
+	defer closer()
+
+	metadataBytes, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		errStr := fmt.Sprintf("could not read body %v", err)
+		log.WithError(err).Errorf(errStr)
+		response.NewErrorResponse(http.StatusUnprocessableEntity, errStr).Write(w)
+		return
+	}
+	chartMetadata := &chart.Metadata{}
+	err = json.Unmarshal(metadataBytes, chartMetadata)
+	if err != nil {
+		errStr := fmt.Sprintf("could not read body %v", err)
+		log.WithError(err).Errorf(errStr)
+		response.NewErrorResponse(http.StatusUnprocessableEntity, errStr).Write(w)
+		return
+	}
+
+	var charts []*models.Chart
+	if err := db.C(chartCollection).Find(bson.M{"name": chartMetadata.Name}).All(&charts); err != nil {
+		log.WithError(err).Errorf("could not find the given chart with id %s", chartMetadata.Name)
+		// continue to return empty list
+	}
+
+	latest := []models.ChartLatest{}
+	for _, c := range charts {
+		if isChartContained(c, chartMetadata) {
+			// We rely that versions are stored in order (newer first) to return the latest
+			latest = append(latest, models.ChartLatest{
+				Chart:  c.Name,
+				Latest: c.ChartVersions[0].Version,
+				Repo:   c.Repo.Name,
+			})
+		}
+	}
+
+	latestRaw, err := json.Marshal(latest)
```

Comment:
Do you need to do this marshaling? In other handlers we seem to do something like `response.NewDataResponse(latest).Write(w)`, which also seems to add the `data` key as described in the https://jsonapi.org/ spec. Any reason for doing it differently here?